sled-agent/src/bootstrap/rack_ops.rs

```diff
@@ -75,7 +76,14 @@ impl RssAccess {
                     }
                     Err(TryRecvError::Empty) => {
                         // Initialization task is still running
-                        RackOperationStatus::Initializing { id }
+                        // Check the step channel, and if it is different
+                        // than what we have, update what we have.
+                        // Ignore errors
+                        let cur_step = step_rx.borrow();
+                        if *step != *cur_step {
+                            *step = *cur_step;
+                        }
+                        RackOperationStatus::Initializing { id, step: *step }
```

Comment:
Small nit - since we always set `*step` to `*cur_step` (except if they're the same... in which case it would also be fine to set it?), could this all be condensed to:

```suggestion
                        RackOperationStatus::Initializing { id, step: *step_rx.borrow() }
```

Reply:
done